h/conftest.py

```diff
@@ -69,6 +69,36 @@ def settings():
     return settings
 
 
+@pytest.fixture(scope='session', autouse=True)
+def setup_database(settings):
+    """Set up the database connection and create tables."""
+    engine = db.make_engine(settings)
+    db.bind_engine(engine, should_create=True, should_drop=True)
+    api_db.use_session(db.Session)
+
+
+@pytest.fixture(autouse=True)
+def database_session(request, monkeypatch):
+    """
+    Prepare the SQLAlchemy session object.
+
+    We enable fast repeatable database tests by setting up the database only
+    once per session (see :func:`setup_database`) and then wrapping each test
+    function in a SAVEPOINT/ROLLBACK TO SAVEPOINT within the transaction.
+    """
+    savepoint = transaction.savepoint()
+
+    def _cleanup():
+        db.Session.remove()
```

Comment:
Since `db.Session.remove` has been monkey-patched to be a no-op below, I think this line is a no-op?